src/main/scala/com/astrolabsoftware/sparkfits/FitsLib.scala

```diff
@@ -208,6 +208,7 @@ object FitsLib {
     val blockHeader = if (conf.get(hdfsPath + "_header") != null) {
       retrieveHeader
     } else readFullHeaderBlocks
+    val fullHeaderBlock = blockHeader
```

Comment:
Why do you define a new variable? One could just use `blockHeader` in `getHDUFullHeader`.